src/include/sof/dma.h

```diff
@@ -85,10 +85,13 @@
 /* DMA copy flags */
 #define DMA_COPY_PRELOAD	BIT(0)
 
+/* DMA preload threshold */
+#define DMA_PRELOAD_THRESHOLD(x)	((x) * 2)
+
 /* We will use this macro in cb handler to inform dma that
  * we need to stop the reload for special purpose
  */
-#define DMA_RELOAD_END	0
+#define DMA_RELOAD_END	0xFFFFFFFE
```

Comment:
this is odd.
The only related change I see is this:
		if (next.size == DMA_RELOAD_END) {
		if (next.size == DMA_RELOAD_END)

But it's not clear what impact the change for this define has? Is this intentional and does this belong in this timer-scheduling change or to a different patch?

Reply:
Getting current available data from DMA can be 0, which is valid in some situations e.g. work_queue tick happened right after starting DMA. Keeping DMA_RELOAD_END as 0 would stop DMA.